staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -126,6 +126,22 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 			return false, 0
 		}
 
+		// Favor this approach over checking with apierrors.IsForbidden since the typed error is needed in the watch
+		// error sent below.
```

Comment:
I don't understand this comment. 
What is your code different from what `IsForbidden` is doing underneath?

IsForbidden underneath is calling reasonAndCodeForError, which does exactly the same thing?

Reply:
@wojtek-t it's not different. It's just that we need the typed error coming from `status.Status()` in the `watch.Event` object created below. So rather than call `IsForbidden` and then if true, repeating what `IsForbidden` is doing to be able to call `.Status()`, this just does it once.

Reply:
It wouldn't be repeating - I think the following would actually be much simpler to follow:
```
if apierrors.IsForbidden(err) {
  klog.ErrorS(...)
  statusErr := err.(apierrors.APIStatus).Status()
```

is cleaner. IsForbidden can't return true, if err is not an APIStatus

Reply:
@wojtek-t good point! I fixed it in the latest push.